src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -344,10 +344,20 @@ public List<DataFile> saveAndAddFilesToDataset(DatasetVersion version,
                     try {
                         StorageIO<DvObject> dataAccess = DataAccess.getStorageIO(dataFile);
                         //Populate metadata
-                        dataAccess.open(DataAccessOption.READ_ACCESS);
-                        // (the .open() above makes a remote call to check if 
-                        // the file exists and obtains its size)
-                        confirmedFileSize = dataAccess.getSize();
+                        
+                        // There are direct upload sub-cases where the file size 
+                        // is already known at this point. For example, direct uploads
+                        // to S3 that go through the jsf dataset page. Or the Globus
+                        // uploads, where the file sizes are looked up in bulk on 
+                        // the completion of the remote upload task. 
+                        if (dataFile.getFilesize() > 0) {
```

Comment:
Can we have 0 length files?

Reply:
Oh, so that line wasn't really breaking support for 0-size files - it was only potentially forcing Dataverse to make extra lookups of the size of 0-size files unnecessarily...